uit/gui_tools/file_browser.py

```diff
@@ -416,31 +416,23 @@ def parent(self):
     def _get_metadata(self):
         if not self.is_absolute():
             self._str = str(self.uit_client.HOME / self)
-        try:
-            ls_text = self.uit_client.call(f'ls -ldL {self.as_posix()}')  # -L dereferences symlinks
-        except UITError:
-            raise ValueError(f'Invalid file path {self.as_posix()}')
-        self._is_dir = False
-        self._is_file = False
-        self._ls = None
 
         if self.name == '':
             # If I don't have a name, don't consider myself a directory or file.
             # Without this, the first directory chosen would break with a grayed-out file listing.
             # It has something to do with loading the home directory for the first time.
             return
 
-        if ls_text.startswith('d'):
-            self._is_dir = True
-            self._ls = self.uit_client.list_dir(self.as_posix())
-            if 'error' in self._ls:
-                # Try our own 'ls -l' if UIT+ returns any error, which happens if there is a broken symlink
-                self._ls = self.parse_list_dir(self.as_posix())
-        elif ls_text.startswith('-'):
+        self._is_dir = True
+        self._is_file = False
+        self._ls = self.uit_client.list_dir(self.as_posix())
+        if self._ls.get('error') == 'File supplied is not a directory.':
+            self._is_dir = False
             self._is_file = True
             self._ls = ''
-        else:
-            raise ValueError(f'Invalid file path {self.as_posix()}')
```

Comment:
I could never reach this exception. If anything goes wrong, UIT+ returns an error which still goes over to self.parse_list_dir(). I have not figured out how to break it; I can still navigate to directories with broken symlinks or no permissions.